frontend/src/components/Pages/FrontPage/MissionOverview/ScheduleMissionDialog.tsx

```diff
@@ -40,9 +44,20 @@ const StyledMissionSection = styled.div`
 
 export const ScheduleMissionDialog = (props: IProps): JSX.Element => {
     const [isDialogOpen, setIsDialogOpen] = useState<boolean>(false)
+    const [isNoEchoMissionsDialogOpen, setIsNoEchoMissionsDialogOpen] = useState<boolean>(false)
     const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false)
-    const [isAssetValid, setIsAssetValid] = useState<boolean>(false)
+    const [isScheduleMissionsPressed, setIsScheduleMissionsPressed] = useState<boolean>(false)
     const anchorRef = useRef<HTMLButtonElement>(null)
+    const { asset } = useAssetContext()
+
+    useEffect(() => {
+        if (!props.isLoadingEchoMissions && isScheduleMissionsPressed) {
```

Comment:
What is the point of this effect? just to check if a dialog is open?

Reply:
To open the dialog when the loading is completed